main.tf

```diff
@@ -1,16 +1,107 @@
+locals {
+  tags = merge(
+    var.tags,
+    {
+      terraform_module = basename(abspath(path.module))
+    }
+  )
+}
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "1.0.0"
+resource "aws_cloudwatch_log_group" "vpn" {
+  name_prefix       = "vpn-${var.name}"
+  retention_in_days = var.log_retention_days
+  tags              = local.tags
+}
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+resource "aws_cloudwatch_log_stream" "vpn" {
+  name           = "vpn-${var.name}"
+  log_group_name = aws_cloudwatch_log_group.vpn.name
 }
 
-locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+resource "aws_iam_saml_provider" "this" {
+  count = var.saml_metadata_document != null ? 1 : 0
+
+  name                   = var.name
+  saml_metadata_document = var.saml_metadata_document
+}
+
+resource "aws_ec2_client_vpn_endpoint" "this" {
+  description            = "Client VPN"
+  client_cidr_block      = var.client_cidr_block
+  server_certificate_arn = var.server_certificate_arn
+  split_tunnel           = var.split_tunnel_enabled
+  tags                   = local.tags
+
+  authentication_options {
+    type              = "federated-authentication"
+    saml_provider_arn = try(aws_iam_saml_provider.this[0].arn, var.saml_provider_arn)
+  }
+
+  connection_log_options {
+    enabled               = true
+    cloudwatch_log_group  = aws_cloudwatch_log_group.vpn.name
+    cloudwatch_log_stream = aws_cloudwatch_log_stream.vpn.name
+  }
+}
+
+resource "aws_security_group" "this" {
+  name_prefix = var.name
+  description = "Client VPN network associations"
+  tags        = var.tags
+  vpc_id      = var.vpc_id
+
+  ingress {
```

Comment:
we shouldn't be doing inline SGs anymore.

Reply:
I'm not sure I agree, since inline SG rules let us prevent drift, but I can see not wanting this hard-coded in the module.